pkg/dhpsi/receiver.go

```diff
@@ -112,6 +112,7 @@ func (s *Receiver) Intersect(ctx context.Context, n int64, identifiers <-chan []
 					// we can match this local identifier with one received
 					// from the sender
 					matchedIDs <- i
+					delete(remoteIDs, p)
```

Comment:
I wonder if simply setting `remoteIDs[p] = false` would be less costly than deleting a bucket from the map.